datashader/core.py

```diff
@@ -138,23 +138,35 @@ class Canvas(object):
     plot_width, plot_height : int, optional
         Width and height of the output aggregate in pixels.
     x_range, y_range : tuple, optional
-        A tuple representing the bounds inclusive space ``[min, max]`` along
-        the axis.
+        A tuple representing the bounds inclusive space ``[min, max]`` along the
+        axis. These will be calculated and cached by datashader during the first
+        aggregation, if not provided. The cached values may be
+        invalidated/recalculated by providing the ``recalc_ranges`` keyword
+        argument to the ``Canvas.points()`` and/or ``Canvas.line()`` methods.
     x_axis_type, y_axis_type : str, optional
         The type of the axis. Valid options are ``'linear'`` [default], and
         ``'log'``.
+    cache_ranges : bool, optional
+        Whether to cache x_range and y_range variables when they get
+        updated. Default is ``True``. This provides a speedup by relying on the
+        x and y data remaining unchanged between data aggregations. If the x or
+        y data may have changed, one may either set this option to ``False``, or
+        provide the ``recalc_ranges=True`` when calling ``Canvas.points()``
+        and/or ``Canvas.line()``.
     """
     def __init__(self, plot_width=600, plot_height=600,
                  x_range=None, y_range=None,
-                 x_axis_type='linear', y_axis_type='linear'):
+                 x_axis_type='linear', y_axis_type='linear',
+                 cache_ranges=True):
         self.plot_width = plot_width
         self.plot_height = plot_height
-        self.x_range = tuple(x_range) if x_range is not None else x_range
-        self.y_range = tuple(y_range) if y_range is not None else y_range
+        self.x_range = tuple(x_range) if x_range is not None else None
+        self.y_range = tuple(y_range) if y_range is not None else None
```

Comment:
I guess you found this way of writing it clearer (as it would seem to have no effect)?  I personally would maybe prefer:

```
self.x_range = None if x_range is None else tuple(x_range)
self.y_range = None if y_range is None else tuple(y_range)
```